MailBounceDetector.Tests/BounceDetectorTests.cs

```diff
@@ -44,6 +44,51 @@ public void ExchangeNonExistingMailbox()
             Assert.Equal("failed", result.Action);
         }
 
+        [Fact]
+        public void ExchangeDelay()
+        {
+            var message = MimeMessage.Load(OpenFixture("bounce_exchange_delay.eml"));
+
+            var result = BounceDetector.Detect(message);
+
+            Assert.True(result.IsBounce);
+            Assert.False(result.IsHard);
+            Assert.True(result.IsSoft);
+            Assert.Equal("4 Persistent Transient Failure", result.PrimaryStatus.ToString());
+            Assert.Equal("4 Network and Routing Status", result.SecundaryStatus.ToString());
+            Assert.Equal("47 Delivery time expired", result.CombinedStatus.ToString());
+            Assert.Null(result.RemoteMta);
+            Assert.Equal("mail.example.com", result.ReportingMta);
+            Assert.Equal("rui.lopes@example.com", result.FinalRecipient);
+            Assert.Equal("<Q8P3DY5EH0U4.ERW6TCCK5TMF3@example.com>", result.UndeliveredMessageId);

```

Comment:
```suggestion
            Assert.Equal("Q8P3DY5EH0U4.ERW6TCCK5TMF3@example.com", result.UndeliveredMessageId);
```

Tests fail otherwise.

I fixed it in my fork: https://github.com/rklec/MailBounceDetector/commit/330f3ab280f8bf4b4484aa99f179197a7168b943